block/src/Pos/Block/Logic/Internal.hs

```diff
@@ -233,7 +233,13 @@ rollbackBlocksUnsafe bsc scb toRollback = do
 toTxpBlock
     :: HasConfiguration
     => Block -> TxpBlock
-toTxpBlock = bimap convertGenesis (convertMain mbTxPayload)
+toTxpBlock block =  case block of
```

Comment:
I know it's WIP, but I want to say that I expect you to get rid of some duplicated code here as well.
Specifically, you can define `toComponentBlock :: (MainBlock -> payload) -> Block -> ComponentBlock payload` and then define `toTxpBlock = toComponentBlock (view mainBlockTxPayload)`.

Reply:
And then `toUpdateBlock = toComponentBlock (view mainBlockUpdatePayload)`, etc.

Reply:
This signature kinda confuses me a bit `toComponentBlock :: (MainBlock -> payload) -> Block -> ComponentBlock payload` - why do I need `Block` in there ?

Reply:
Because `Block` contains the data you need 🤷‍♂️ 
Look at current version of `toTxpBlock`, it also takes `Block` as an argument.
Do you have something different in mind?

Reply:
I think I got you, I made everything compile up to a point but there are more places where `Either` was used instead this new `ComponentBlock` type so I am continuing my work